src/main/java/org/sonar/scm/git/blame/GraphNodeFactory.java

```diff
@@ -58,8 +70,30 @@ public StatefulCommit create(TreeWalk treeWalk, RevCommit commit) throws IOExcep
 
       treeWalk.getObjectId(idBuf, 0);
       files.add(new FileCandidate(treeWalk.getPathString(), treeWalk.getPathString(), idBuf.toObjectId()));
+
+    }
+    return new CommitGraphNode(commit, files);
+  }
+
+  public GraphNode createForWorkingDir(TreeWalk treeWalk, RevCommit parentCommit) throws IOException {
+    Objects.requireNonNull(parentCommit);
+    List<FileCandidate> files = new ArrayList<>();
+
+    treeWalk.setRecursive(true);
+    treeWalk.reset();
+    treeWalk.addTree(new FileTreeIterator(repository));
+
+    while (treeWalk.next()) {
+      if (filePathsToBlame != null && !filePathsToBlame.contains(treeWalk.getPathString())) {
+        continue;
+      }
+      if (!isFile(treeWalk.getRawMode(0))) {
+        continue;
+      }
+      files.add(new FileCandidate(treeWalk.getPathString(), treeWalk.getPathString(), ObjectId.zeroId()));
+
     }
-    return new StatefulCommit(commit, files);
+    return new WorkDirGraphNode(parentCommit, files);
```

Comment:
If no files found, it is worth to fallback on CommitGraphNode to avoid processing a useless GraphNode later?

Reply:
If no files are found, there's nothing to blame and the graph iteration would stop after the first iteration.
Maybe we could change `BlameGenerator#prepareStartCommit` to not even push that node in that case